crates/matrix-sdk-ui/src/timeline/traits.rs

```diff
@@ -121,6 +123,10 @@ impl RoomDataProvider for Room {
         (**self).clone_info().room_version_or_default()
     }
 
+    fn client_info(&self) -> BoxFuture<'_, ClientInfo> {
+        async move { self.client_info().await }.boxed()
+    }
```

Comment:
No the struct method has priority other the Trait method. If I wanted to call the trait method I'd have to explicitely do `RoomDataProvider::crypto_context_info(&room)` or maybe if the type was something like `&dyn RoomDataProvider` and not Room.

To avoid confusion maybe we could rename one of the method (adding a `get_` prefix in the trait)?
But it is not the same occurence of that in this file, see `load_user_receipt`